File: TransCoder_tokenized_test_set_functions/java/MODIFY_ARRAY_TO_MAXIMIZE_SUM_OF_ADJACENT_DIFFERENCES.java
static int maximumDifferenceSum ( int arr [ ], int N ) {
  int dp [ ] [ ] = new int [ N ] [ 2 ];
  for ( int i = 0;
  i < N;
  i ++ ) dp [ i ] [ 0 ] = dp [ i ] [ 1 ] = 0;
  for ( int i = 0;
  i < ( N - 1 );
  i ++ ) {
    dp [ i + 1 ] [ 0 ] = Math . max ( dp [ i ] [ 0 ], dp [ i ] [ 1 ] + Math . abs ( 1 - arr [ i ] ) );
    dp [ i + 1 ] [ 1 ] = Math . max ( dp [ i ] [ 0 ] + Math . abs ( arr [ i + 1 ] - 1 ), dp [ i ] [ 1 ] + Math . abs ( arr [ i + 1 ] - arr [ i ] ) );
  }
  return Math . max ( dp [ N - 1 ] [ 0 ], dp [ N - 1 ] [ 1 ] );
}

File: TransCoder_tokenized_test_set_functions/python/MODIFY_ARRAY_TO_MAXIMIZE_SUM_OF_ADJACENT_DIFFERENCES.py
def maximumDifferenceSum ( arr, N ) :
    dp = [ [ 0, 0 ] for i in range ( N ) ]
    for i in range ( N ) :
        dp [ i ] [ 0 ] = dp [ i ] [ 1 ] = 0
    for i in range ( N - 1 ) :
        dp [ i + 1 ] [ 0 ] = max ( dp [ i ] [ 0 ], dp [ i ] [ 1 ] + abs ( 1 - arr [ i ] ) )
        dp [ i + 1 ] [ 1 ] = max ( dp [ i ] [ 0 ] + abs ( arr [ i + 1 ] - 1 ), dp [ i ] [ 1 ] + abs ( arr [ i + 1 ] - arr [ i ] ) )
    return max ( dp [ N - 1 ] [ 0 ], dp [ N - 1 ] [ 1 ] )
